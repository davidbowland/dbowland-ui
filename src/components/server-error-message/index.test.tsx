import React from 'react'
import '@testing-library/jest-dom'
import { render } from '@testing-library/react'

import ServerErrorMessage from './index'

describe('Server error message component', () => {
  const title = 'server-error-message'
  const children = <span>Nothing to see here</span>

  test('Ensure snapshot match', () => {
    const { container } = render(<ServerErrorMessage title={title}>{children}</ServerErrorMessage>)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('Ensure title in output', () => {
    const { getByText } = render(<ServerErrorMessage title={title}>{children}</ServerErrorMessage>)
    expect(getByText(title)).toBeInTheDocument()
  })

  test('Ensure link to home', () => {
    const { container } = render(<ServerErrorMessage title={title}>{children}</ServerErrorMessage>)
    const anchorArray: HTMLAnchorElement[] = Array.from(container.querySelectorAll('a'))
    expect(anchorArray.filter((anchor) => new URL(anchor.href).pathname === '/').length).toBe(1)
  })
})
