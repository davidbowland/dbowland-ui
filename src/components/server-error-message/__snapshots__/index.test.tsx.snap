// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server error message component Ensure snapshot match 1`] = `
<main
  class="sc-dlnjwi gnIxof"
>
  <h1
    class="sc-gtsrHT elXEts"
  >
    server-error-message
  </h1>
  <p
    class="sc-bdnxRM fSObbe"
  >
    <span>
      Nothing to see here
    </span>
    <br />
    <br />
    <a
      href="/"
    >
      Go home
    </a>
    .
  </p>
</main>
`;
